### PR TITLE
[#2361] files/folders which directly match svn and cvs keywords are ignored

### PR DESCRIPTION
**Pull request by denis631**
Related to https://github.com/LightTable/LightTable/issues/2361

### CODE REVIEW

File: src/lt/objs/files.cljs [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 (def ignore-pattern
   "Regex pattern consisting of files, folders, etc... to ignore."
-  #"(^\..*)|\.class$|target/|svn|cvs|\.git|\.pyc|~|\.swp|\.jar|.DS_Store")
+  #"(^\..*)|\.class$|target/|^svn$|^cvs$|\.git|\.pyc|~|\.swp|\.jar|.DS_Store")
```
Comment (Mouvedia) [inline src/lt/objs/files.cljs]: The part that you have modified should be `^\.svn$|^CVS$`.

**Comment by Mouvedia** [comment]
@denis631 read about SVN_ASP_DOT_NET_HACK

tl;dr: replace `^\.svn$` by `^[_.]svn$`

**Comment by denis631** [comment]
@Mouvedia don't I have to escape the dot? meaning: ```^[_\.]svn$```

**Comment by Mouvedia** [comment]
> don't I have to escape the dot?

No you don't, not in a character class.

File: src/lt/objs/files.cljs [hunk 2]
```diff
@@ -31,7 +31,7 @@
 
 (def ignore-pattern
   "Regex pattern consisting of files, folders, etc... to ignore."
-  #"(^\..*)|\.class$|target/|^svn$|^cvs$|\.git|\.pyc|~|\.swp|\.jar|.DS_Store")
+  #"(^\..*)|\.class$|target/|^[_\.]svn$|^cvs$|\.git|\.pyc|~|\.swp|\.jar|.DS_Store")
```
Comment (Mouvedia) [inline src/lt/objs/files.cljs]: cvs > CVS

**Comment by Mouvedia** [comment]
Now let's fix #1153 while we are at it: `^\.git$`.  
The `$` is required for `.gitignore`.

**Comment by Mouvedia** [comment]
@denis631 you forgot mercurial: `.hg`

**Comment by Mouvedia** [comment]
LGTM, please squash

**Comment by denis631** [comment]
Shouldn't we group somehow the ignore-pattern. I mean we have some patterns regarding version control, other are language specific, like .pyc, etc.

Shouldn't language plugin add this functionality? Or at least be able to?

**Comment by Mouvedia** [comment]
It should, in the same way that the syntax highlighting changes depending on the file extension but that would require to have some kind of project context.

There are more pressing issues though: https://github.com/LightTable/LightTable/projects/4

**Comment by Mouvedia** [comment]
@sbauer322 LGTM

**Comment by sbauer322** [comment]
This seems fine. Thank you for the PR, @denis631, and for your assistance, @Mouvedia!

Apologies for the delay.

**Pull request merged**